.ci/release/Jenkinsfile

```diff
@@ -0,0 +1,235 @@
+      options { skipDefaultCheckout () }
+      stages{
+        // 1. Check to see if oss.sonatype.org is up. We throw an exception and fail the job if there is no response.
+        stage('Check oss.sonatype.org') {
+          steps {
+            // If this fails, an exception should be thrown and execution will halt
+            httpRequest(url: "https://oss.sonatype.org/#welcome")
+          }
+        }
+        // 2. Review project version.
+        stage('Review project version') {
+          steps {
+            script {
+              // Show the select input modal
+              def INPUT_PARAMS = input message: 'Do you wish to update the version?', ok: 'Next', parameters: [choice(name: 'UPDATE_CHOICE', choices:['Yes', 'No'], description: "This will run `mvn release:update-versions`")]
+              if (INPUT_PARAMS == 'Yes') {
+                  dir("${BASE_DIR}") {
+                    sh mvn release:update-versions
+                  }
+              } else {
+                  echo 'Skipping version update'
+              }
+            }
+          }
+        }
+        // 3. Execute the release Jenkins job on the internal ci server.
+            // For now we assume that the release has been pushed to staging already on the internal CI
+        // 4. Login to oss.sonatype.org, go to Staging Repositories, close and release the staged artifacts.
+        stage('Nexus release') {
+          steps {
+            script {
+              def spid = getVault('nexus')["staging-profile-id"]
+              dir("${BASE_DIR}"){
+                withEnvMask(vars: [[var: "SPID", password: spid]]){
+                  def foundStagingId = nexusFindStagingId(stagingProfileId: "${SPID}", groupId: "co.elastic.apm")
+                  nexusCloseStagingRepository(stagingProfileId: "${SPID}", stagingId: foundStagingId)
+                  nexusReleaseStagingRepository(stagingProfileId: "${SPID}", stagingId: foundStagingId)
+                }
+
+              }
+            }
+          }
+        }
+        // 5. Fetch and checkout the latest tag e.g. git fetch origin
+          // We already have this :)
+        // 6. If this was a major release, create a new branch for the major.
+        stage('Branch creation') {
+          steps {
+            dir("${BASE_DIR}") {
+              script {
+                def isMajor = sh(script: 'git tag|tail -1|cut -f2-3 -d "."|{ read ver; test $ver == "0.0"; }', returnStatus: true)
+                if (isMajor == 0) {
+                  // We need to create a new branch. First get the name of the branch
+                  def newBranchName = sh(script: 'git tag|tail -1|sed s/v//|cut -f1 -d "."|awk \'{print $1".x"}\'', returnStdout: true)
+                  echo "This appears to be a major version. We are creating a new branch for $newBranchName in the apm-agent-java repo on GitHub."
+                  // Now we need to branch
+                  sh(script: "git checkout -b $newBranchName")
```

Comment:
What do you think to move this logic to a shell script?

Reply:
Also, as an alternative, it might be easier to get project version from maven `pom.xml` rather than parsing it from git.

Reply:
Great idea! I have [implemented this upstream](https://github.com/elastic/apm-pipeline-library/pull/432) and will update this pipeline to use the new functionality.